readme.md

```diff
@@ -47,6 +47,21 @@ Java files will be generated under `where/to/generate/java/files/src/main/java/s
 
 To debug, add `--java.debugger` to the argument list. The JVM will suspend at the beginning of the execution. Then attach a remote debugger in your IDE to `localhost:5005`. **Make sure you detach the debugger before killing the AutoRest process. Otherwise it will fail to shutdown the JVM and leave it orphaned. (which can be killed in the Task Manager)**
 
+# Recommended Settings
+
+- `client-logger: true`, Exception is required to be thrown via `ClientLogger`.
+- `use-default-http-status-code-to-exception-type-mapping: true`, use subclass of `HttpResponseException` defined in azure-core, when service returns unexpected status code.
+- `generic-response-type: true`, use `ResponseBase<H, T>` for response with HTTP headers.
+- `output-model-immutable: true`, make model classes immutable, if model is used only in response.
+
+Settings for handwritten SDK:
+
+- `generate-client-as-impl: true`, generate Client in implementation package, for customization.
+- `models-subpackage: implementation.models`, generate model classes in implementation package.
+- `custom-types-subpackage: models`, generate selected model classes in models package.
+- `service-interface-as-public: true`, generate the interface consumed by `RestProxy` as public to resolve issues with `SecurityManager`.
```

Comment:
We should make the default value for this setting `true`. We should also make the service interface public (in the implementation package).

Reply:
I will try to make it default to be true, as usually this client is in implementation package.